aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -398,9 +405,13 @@ def presubmit(self, folder):
 
         # Set resources, also with get_default_mpiprocs_per_machine
         resources = self.node.get_option('resources')
-        def_cpus_machine = computer.get_default_mpiprocs_per_machine()
-        if def_cpus_machine is not None:
-            resources['default_mpiprocs_per_machine'] = def_cpus_machine
+        default_mpiprocs_per_machine = computer.get_default_mpiprocs_per_machine()
```

Comment:
The general doubt I have here is if this is a general logic, or still specific to some scheduler types. Do we have tests for a parenv job resource, or can we ask someone (@zhubonan, @atztogo, ...) to check if this new logic now works also for them?

Reply:
I think you are right and this will fail. Even if the user does not explicitly specify `num_mpiprocs_per_machine` and `tot_num_mpiprocs`, if the computer has `default_mpiprocs_per_machine` defined (which I think currently is almost always the case, or at least, `verdi computer setup` suggests to set it), then `num_mpiprocs_per_machine` will be added to the `resources` and since I changed `ParEnvJobResource.validate_resources` to raise when it gets unsupported keywords, this will fail.

I am not sure where and how to fix it. Easy hack would be to simply ignore this keyword in the `ParEnvJobResource` but is not the *correct* solution. If specifying a number of machines for a `ParEnvJobResource` based scheduler never makes sense, maybe we should not even have computers that are configured with such a scheduler, define a `num_mpiprocs_per_machine`.